### ISSUE TITLE
Only embedded one subtitle language after converting

### ISSUE THREAD

**Issue by bagusnl** [ytdl-org/youtube-dl#28537]
## Checklist

- [ ] I'm reporting a broken site support issue
- [x] I've verified that I'm running youtube-dl version **2021.03.25**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar bug reports including closed ones
- [x] I've read bugs section in FAQ

## Verbose log

```
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-f', 'bestvideo[vcodec!*=av01]+bestaudio', '--proxy', '', '--geo-bypass', '--no-playlist', '--write-sub', '--all-subs', '--sub-format', 'ttml', '--convert-subs', 'ass', '--embed-subs', '--write-thumbnail', '--embed-thumbnail', '--merge-output-format', 'mkv', '-v', '--console-title', 'https://www.youtube.com/watch?v=Ud73fm4Uoq0']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, pref cp1252
[debug] youtube-dl version 2021.03.25
[debug] Python version 3.9.2 (CPython) - Windows-10-10.0.19041-SP0
[debug] exe versions: ffmpeg 4.3.2-2021-02-20-essentials_build-www.gyan.dev, ffprobe 4.3.2-2021-02-20-essentials_build-www.gyan.dev
[debug] Proxy map: {}
[youtube] Ud73fm4Uoq0: Downloading webpage
[youtube] Downloading just video Ud73fm4Uoq0 because of --no-playlist
[info] Writing video subtitles to: 【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.en.ttml
[info] Writing video subtitles to: 【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.ja.ttml
[youtube] Ud73fm4Uoq0: Downloading thumbnail ...
[youtube] Ud73fm4Uoq0: Writing thumbnail to: 【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.webp
[debug] Invoking downloader on 'https://r3---sn-uxa3vhnxa-jb3e7.googlevideo.com/videoplayback?expire=1616660078&ei=DfJbYNr4O5j84-EPkMqwwAE&ip=114.124.130.7&id=o-AAaMTuNQOrwCn2nI51WNRw2xrdGv-BZEq2lqhtzMnZWY&itag=137&aitags=133%2C134%2C135%2C136%2C137%2C160%2C242%2C243%2C244%2C247%2C248%2C278%2C394%2C395%2C396%2C397%2C398%2C399&source=youtube&requiressl=yes&mh=uv&mm=31%2C26&mn=sn-uxa3vhnxa-jb3e7%2Csn-i3belnl6&ms=au%2Conr&mv=m&mvi=3&pl=18&initcwndbps=311250&vprv=1&mime=video%2Fmp4&ns=LUds0pWadLvwrtK6WIh6h3gF&gir=yes&clen=45051231&dur=220.750&lmt=1616450254862397&mt=1616638362&fvip=3&keepalive=yes&fexp=24001373%2C24007246&c=WEB&txp=8316222&n=cJu6rEGPxhSpn_oMnzej7&sparams=expire%2Cei%2Cip%2Cid%2Caitags%2Csource%2Crequiressl%2Cvprv%2Cmime%2Cns%2Cgir%2Cclen%2Cdur%2Clmt&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRgIhAIF-r1Z-69y19gd7Z4kC5cF7QcSes0ECRwLFTm643gxAAiEAs274O65jWcIf2YQsj-gwiL-rxrkalJ46jyADZA4T5Zk%3D&sig=AOq0QJ8wRAIgBaqubuL9AzffBjc0QLydb9iS0zn8xdWCNboEOUcwqdECIDROaay4A7IS3JZB9kX--PWLkPAo5L5DJeMFQa6rX8aw'
[download] Destination: 【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.f137.mp4
[download] 100% of 42.96MiB in 00:22
[debug] Invoking downloader on 'https://r3---sn-uxa3vhnxa-jb3e7.googlevideo.com/videoplayback?expire=1616660078&ei=DfJbYNr4O5j84-EPkMqwwAE&ip=114.124.130.7&id=o-AAaMTuNQOrwCn2nI51WNRw2xrdGv-BZEq2lqhtzMnZWY&itag=251&source=youtube&requiressl=yes&mh=uv&mm=31%2C26&mn=sn-uxa3vhnxa-jb3e7%2Csn-i3belnl6&ms=au%2Conr&mv=m&mvi=3&pl=18&initcwndbps=311250&vprv=1&mime=audio%2Fwebm&ns=LUds0pWadLvwrtK6WIh6h3gF&gir=yes&clen=3653445&dur=220.781&lmt=1616450241034066&mt=1616638362&fvip=3&keepalive=yes&fexp=24001373%2C24007246&c=WEB&txp=8311222&n=cJu6rEGPxhSpn_oMnzej7&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cvprv%2Cmime%2Cns%2Cgir%2Cclen%2Cdur%2Clmt&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRgIhAOyax5zIzfY1ch-0wVnnxv6IcAkp1rdpZ_a3VP6p5CIfAiEAxlpjrV0zNn4b94x9urATzwmr5jgbTiLBEI-SYSXDMuw%3D&sig=AOq0QJ8wRQIgSfGYcGM2MvFKulKSIam8laiHFMMsUFUHLSfLwVoPTIwCIQDl8-QulpvQ115ui3x6qiFZ0TUNXAm_1JBGjGau3XRXRw=='
[download] Destination: 【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.f251.webm
[download] 100% of 3.48MiB in 00:02
[ffmpeg] Merging formats into "【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.mkv"
[debug] ffmpeg command line: ffmpeg -y -loglevel "repeat+info" -i "file:【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.f137.mp4" -i "file:【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.f251.webm" -c copy -map "0:v:0" -map "1:a:0" "file:【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.temp.mkv"
Deleting original file 【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.f137.mp4 (pass -k to keep)
Deleting original file 【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.f251.webm (pass -k to keep)
[ffmpeg] Converting subtitles
WARNING: You have requested to convert dfxp (TTML) subtitles into another format, which results in style information loss
[debug] ffmpeg command line: ffmpeg -y -loglevel "repeat+info" -i "file:【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.en.srt" -f ass "file:【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.en.ass"
WARNING: You have requested to convert dfxp (TTML) subtitles into another format, which results in style information loss
[debug] ffmpeg command line: ffmpeg -y -loglevel "repeat+info" -i "file:【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.ja.srt" -f ass "file:【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.ja.ass"
Deleting original file 【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.en.ttml (pass -k to keep)
Deleting original file 【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.en.srt (pass -k to keep)
Deleting original file 【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.ja.ttml (pass -k to keep)
Deleting original file 【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.ja.srt (pass -k to keep)
[ffmpeg] Embedding subtitles in '【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.mkv'
[debug] ffmpeg command line: ffmpeg -y -loglevel "repeat+info" -i "file:【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.mkv" -i "file:【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.en.ass" -i "file:【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.ja.ass" -map 0 -c copy -map "-0:s" -map "-0:d" -map "1:0" "-metadata:s:s:0" "language=eng" -map "2:0" "-metadata:s:s:1" "language=jpn" "file:【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.temp.mkv"
Deleting original file 【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.en.ass (pass -k to keep)
Deleting original file 【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.ja.ass (pass -k to keep)
[ffmpeg] Converting thumbnail "【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.webp" to JPEG
[debug] ffmpeg command line: ffmpeg -y -loglevel "repeat+info" -i "file:【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.webp" "-bsf:v" mjpeg2jpeg "file:【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.jpg"
[ffmpeg] Adding thumbnail to "【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.mkv"
[debug] ffmpeg command line: ffmpeg -y -loglevel "repeat+info" -i "file:【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.mkv" -c copy -attach cover.jpg "-metadata:s:t" "mimetype=image/jpeg" "file:【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.temp.mkv"
```

## Description

youtube-dl only embedded one of the subtitle language after converting one subtitle format to another even when it said its downloaded and converted another subtitle language.
Here is MediaInfo dump of the output file
```
General
Unique ID                   : 47465439081717350072895866265216495303 (0x23B582D3D9F96086C526CF4BB05C2EC7)
Complete name               : C:\Users\bagusnl_reg\Downloads\Video\【オリジナル曲】　Palette_常闇トワ　【フルMV】-Ud73fm4Uoq0.mkv
Format                      : Matroska
Format version              : Version 4
File size                   : 46.8 MiB
Duration                    : 3 min 40 s
Overall bit rate            : 1 779 kb/s
Writing application         : Lavf58.45.100
Writing library             : Lavf58.45.100
Cover                       : Yes
ErrorDetectionType          : Per level 1
Attachments                 : cover.jpg

Video
ID                          : 1
Format                      : AVC
Format/Info                 : Advanced Video Codec
Format profile              : High@L4
Format settings             : CABAC / 3 Ref Frames
Format settings, CABAC      : Yes
Format settings, Reference  : 3 frames
Codec ID                    : V_MPEG4/ISO/AVC
Duration                    : 3 min 40 s
Width                       : 1 920 pixels
Height                      : 1 080 pixels
Display aspect ratio        : 16:9
Frame rate mode             : Constant
Frame rate                  : 24.000 FPS
Color space                 : YUV
Chroma subsampling          : 4:2:0
Bit depth                   : 8 bits
Scan type                   : Progressive
Title                       : ISO Media file produced by Google Inc.
Default                     : Yes
Forced                      : No
Color range                 : Limited
Color primaries             : BT.709
Transfer characteristics    : BT.709
Matrix coefficients         : BT.709

Audio
ID                          : 2
Format                      : Opus
Codec ID                    : A_OPUS
Duration                    : 3 min 40 s
Channel(s)                  : 2 channels
Channel layout              : L R
Sampling rate               : 48.0 kHz
Bit depth                   : 32 bits
Compression mode            : Lossy
Delay relative to video     : -14 ms
Language                    : English
Default                     : Yes
Forced                      : No

Text
ID                          : 3
Format                      : ASS
Codec ID                    : S_TEXT/ASS
Codec ID/Info               : Advanced Sub Station Alpha
Duration                    : 3 min 16 s
Compression mode            : Lossless
Language                    : English
Default                     : Yes
Forced                      : No
```
Its missing one of the subtitle language for Japanese

**Comment by kikuyan** [comment]
May I ask, are you using officially released youtube-dl version 2021.03.25?
When I run youtube-dl with your options, I get ```ERROR: Only mp3 and m4a/mp4 are supported for thumbnail embedding for now.```, and the last 2 lines of your verbose log indicate that you are using not yet merged pull request #22659 or derived embedthumbnail.py module.

**Comment by bagusnl** [comment]
> May I ask, are you using officially released youtube-dl version 2021.03.25?
> When I run youtube-dl with your options, I get `ERROR: Only mp3 and m4a/mp4 are supported for thumbnail embedding for now.`, and the last 2 lines of your verbose log indicate that you are using not yet merged pull request #22659 or derived embedthumbnail.py module.

Oh yeah I used that PR for fixing the thumbnail embed not knowing that it could actually break the subtitle embedding module. My bad didn't check it the first time but using the release from this repo resulting expected error for thumbnail embedding but fixed subtitle embedding.

I'm truly sorry for that, and I'm closing this issue

Thanks for bringing that to my attention @kikuyan
